sanguk/app.js

```diff
@@ -33,20 +33,77 @@ app.get("/ping", (req,res)=>{
 })
 
 app.post("/users/signup", async (req,res,next)=>{
-    const {username, password, email, phonenumber } =req.body
+    const {username, userProfileImage } =req.body
 
     await myDataSource.query(
         `INSERT INTO users(
             username,
-            password,
-            email,
-            phonenumber
-        ) VALUES(?, ?, ?, ?)`
-        ,[username, password, email, phonenumber]
+            userProfileImage
+        ) VALUES(?, ?)`
+        ,[username, userProfileImage]
     )
     res.status(201).json({message : "userCreated"})
 })
 
+app.post("/posts/signup", async (req,res,next)=>{
+    const {postingId, postingImageUrl, postingContent} =req.body
+
+    await myDataSource.query(
+        `INSERT INTO posts(
+            postingId,
+            postingImageUrl,
+            postingContent
+        ) VALUES(?, ?, ?)`
+        ,[postingId, postingImageUrl, postingContent]
+    )
+    res.status(201).json({message : "postCreated"})
+})
+
+app.get ('/search/everything',async (req,res)=>{
+    await myDataSource.manager.query(    
+    `SELECT
+        users.id as userId,
+        users.userProfileImage,
+        posts.postingId,
+        posts.postingImageUrl,
+        posts.postingContent
+        FROM users
+        LEFT JOIN posts ON users.id=postingId
+`
+        ,(err, rows)=>{
+            res.status(200).json({data:rows});
+        }
+    )
+})
+
+app.get ('/search/posts_user',async (req,res)=>{
+    await myDataSource.manager.query(    
```

Comment:
```suggestion
app.get ('/users/:userId/posts', async (req,res)=>{
    const { userId } = req.params
    
    await myDataSource.manager.query(    
```
아마 과제가 특정 유저가 작성한 게시글 목록을 조회하는 API 구현이셨을거에요! parameter를 통해 특정할 userId를 받고 데이터베이스에 있는 `'유저들중, 몇 번 유저의, 게시글 들을 조회하는지'` -> `app.get("/users/:userId/posts", . . . ` 와 같이 나타내면 좋을 것 같습니다 !

Reply:
위에 내용을 같이 합쳐서 수정 하였습니다.